guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/CacheSerializer.java

```diff
@@ -1,29 +1,88 @@
 package com.fasterxml.jackson.datatype.guava.ser;
 
-import java.io.IOException;
-
-import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.core.JsonToken;
 import com.fasterxml.jackson.core.type.WritableTypeId;
-import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMember;
 import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import com.fasterxml.jackson.databind.ser.ContextualSerializer;
 import com.fasterxml.jackson.databind.ser.std.StdSerializer;
 
 import com.google.common.cache.Cache;
 
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Serializer for Guava's {@link Cache}.
+ * <p>
+ * Missing features, compared to standard Java Maps:
+ * <ul>
+ *  <li>Sorting of entries</li>
+ * </ul>
+ * 
+ * @since 2.16
+ */
 public class CacheSerializer extends StdSerializer<Cache<?, ?>>
```

Comment:
You may want to instead extend `ContainerSerializer` like standard `MapSerializer` does. It provides some support for building features and can show what is missing.